pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -98,11 +104,19 @@ func (ngm *nvidiaGPUManager) Start() error {
 	if !os.IsNotExist(err) {
 		ngm.defaultDevices = append(ngm.defaultDevices, nvidiaUVMToolsDevice)
 	}
-
-	if err := ngm.discoverGPUs(); err != nil {
-		return err
-	}
-
+	// retry discoverGPUs indefinitely in case drivers have not been installed
+	go func() {
+		var err error
+		for {
+			if err := ngm.discoverGPUs(devicesProcFile, devDirectory); err == nil {
+				return
+			}
+			glog.Errorf("failed to discover Nvidia GPUs: %v", err)
```

Comment:
I'm worried about this as well. Should it be changed to an info message or should I reconsider whether to log at all?